papas/simulation/HelixPropagator.h

```diff
@@ -2,37 +2,27 @@
 #ifndef helixpropagator_h
 #define helixpropagator_h
 #include "papas/detectors/Detector.h"
+#include "papas/simulation/Propagator.h"
 
 namespace papas {
 class PFParticle;
-class HelixPropagator {
+  
+class HelixPropagator : public Propagator {
 /** Class to determine where the (helix) path of a charged particle crosses the detector cyclinders
 */
 public:
   /** Constructor
-   field Magnetic field in Tesla(?) (assumed constant)
   */
-  HelixPropagator(double field);
+  HelixPropagator();
   /**
    Propagate charged particle along Helix to the selected cylinder and store the point
    where the particle crosses the cylinder
-   ptc particle that is to be propagated
-   cyl cylinder to which the particle is to be propagated.
+   @param[in] ptc particle that is to be propagated
+   @param[in] cyl cylinder to which the particle is to be propagated.
+   @param[in] field magnitude of magnetic field
    */
-  virtual void propagateOne(const PFParticle& ptc, const SurfaceCylinder& cyl) const;
+  virtual void propagateOne(const PFParticle& ptc, const SurfaceCylinder& cyl, double field) const override;
```

Comment:
I can do this. However I chose to do it this way because it reduces the dependency of the propagator class on the DetectorElement classes...
Also if you do want this I would like to add it into the Particles3 branch as there are further pending changes then to the way the field is used.